src/frontends/common/include/openvino/frontend/input_model.hpp

```diff
@@ -36,10 +36,20 @@ namespace frontend {
 /// All editing requests affect the model representation that is held behind the scene
 /// successive method calls observe a new graph structure.
 class FRONTEND_API InputModel {
+    std::shared_ptr<void> m_shared_object;
+    std::shared_ptr<InputModel> m_actual;
+    friend class FrontEnd;
+
 public:
-    typedef std::shared_ptr<InputModel> Ptr;
+    using Ptr = std::shared_ptr<InputModel>;
+
+    InputModel(const std::shared_ptr<void>& shared_object, const InputModel::Ptr& actual)
```

Comment:
private?

Reply:
Fixed (removed). In case that std::make_shared doesn't work with private constructors, I've decided to use std::make_shared without arguments, then set these members manually